pkgs/development/tools/konf/default.nix

```diff
@@ -1,22 +1,45 @@
 { lib
 , buildGoModule
 , fetchFromGitHub
+, installShellFiles
+, testers
+, konf
 }:
 
 buildGoModule rec {
   pname = "konf";
-  version = "0.3.0";
+  version = "0.4.0";
 
   src = fetchFromGitHub {
     owner = "SimonTheLeg";
     repo = "konf-go";
     rev = "v${version}";
-    hash = "sha256-n8WJyihpbtci8Q6zUapwrpTrVZVAS53OpnwAsv6w1FY=";
+    hash = "sha256-8TXr/jYMl3NLERtLkm7qG97IL/idz4xxP0g0LEy4/j8=";
   };
 
   vendorHash = "sha256-sB3j19HrTtaRqNcooqNy8vBvuzxxyGDa7MOtiGoVgN8=";
 
-  ldflags = [ "-s" "-w" ];
+  ldflags = [
+    "-s" "-w"
+    "-X github.com/simontheleg/konf-go/cmd.gitcommit=unknown"
+    "-X github.com/simontheleg/konf-go/cmd.builddate=unknown"
+    "-X github.com/simontheleg/konf-go/cmd.gitversion=${version}"
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    export HOME=$(mktemp -d)
+    ln -s $out/bin/konf-go $out/bin/konf
```

Comment:
Done. Let me know of further issues.

Reply:
I lack context here. I'm only updating package because upstream has added version to package. And I thought that it was an usual application. It is also missing `meta.mainProgram` but now, that you say that it follows a different pattern, I don't know what to do. Regarding wrappers, there is `makeWrapper` (many examples available in nixpkgs), is that what you reference?

Reply:
Thanks for your comment. I was lacking context on this `shellwrapper` issue. As I'm not an user of application. I guess it is better for me to close issue and leave it to someone with more context of issue.